xwiki-platform-core/xwiki-platform-oldcore/src/main/java/com/xpn/xwiki/web/DeleteAction.java

```diff
@@ -214,7 +219,7 @@ private void deleteFromRecycleBin(long index, XWikiContext context) throws XWiki
         sendRedirect(response, Utils.getRedirect("view", context));
     }
 
-    private boolean deleteToRecycleBin(XWikiContext context) throws XWikiException
+    private boolean deleteToRecycleBin(XWikiContext context, boolean skipRecycleBin) throws XWikiException
```

Comment:
I find it a bit misleading. You'd think that calling this will skip the recycle bin but in practice it'll only be done based on conditions (permissions).

Reply:
Also the method name is not good since it shouldn't day "delete to recycle bin" since that's not true if the boolean is false or if the user doesn't have the permissions

Reply:
`deleteToRecycleBin` renamed `deleteDocument` (see https://github.com/xwiki/xwiki-platform/pull/1401/files#diff-70855079f42aabba98e87755fa9e71ecR262)